### ISSUE TITLE
Positivity discrepancy between a Set parameter and one detected as parametric by forcing

### ISSUE THREAD

**Issue by gallais** [agda/agda#3778]
```agda
data C₁ (A : Set) : Set where
  c : C₁ A

data C₂ : Set → Set where
  c : ∀ {A} → C₂ A

data D₁ : Set where
  d : C₁ D₁ → D₁

data D₂ : Set where
  d : C₂ D₂ → D₂
```

Even though `C₁` and `C₂` are virtually identical (note that `C₂` is correctly
accepted as a small type because forcing notices that `A` is in fact a
parameter), they behave differently when it comes to positivity.

`D₁` is accepted just fine while `D₂` is rejected with the error

```
D₂ is not strictly positive, because it occurs
in the first argument of C₂
in the type of the constructor d
in the definition of D₂.
```

Bug found by @ichistmeinname [on SO](https://stackoverflow.com/questions/56143587/why-is-my-definition-not-allowed-because-of-strict-positivity)

**Comment by gallais** [comment]
Oh wait, this is a duplicate of #3541, right?

**Comment by gallais** [comment]
Should be fixed by cd25ceeaa35e7df5642917c5931fcbc483bacd60

Edit: indeed it is!
